### ISSUE TITLE
Allow starting a persistent server via plugin

### ISSUE THREAD

**Issue by jpommerening** [aixigo/pdfreactor-loader#6]
Instead of starting a server every time a PDF is rendered, allow for a server that keeps running so the user can use their browser and see the exact same HTML that the `pdfreactor-loader` renders.
The `pdfreactor-loader` should expose a webpack plugin to start a persistent server when `webpack --watch` is used. The loader can then use the running server for additional resources referenced by the HTML and "register" the HTML that it received for a given input document (`resourcePath`).

This should resolve aixigo/freemarker-loader#2 which attempted to solve the use case of displaying PDFreactors HTML input, by writing a file to disk.

**Comment by jpommerening** [comment]
Implemented on `master` (`v0.3.0`)

**Comment by jpommerening** [comment]
## New Feature

You can now use a plugin to share a server between multiple loader executions (for example when watching, or rendering multiple files).

There's a whole chapter in the [README.md](../#built-in-http-server), but here's a short usage example. In this example, resources will be served from `__dirname` (the directory the webpack.config.js is placed in) and rendered files can be inspected at http://localhost:8099/.

```js
// webpack.config.js
module.exports = {
   context: __dirname,
   module: {
      rules: [
         {
            test: /\.html\.pdf$/,
            use: [ 'file-loader', 'pdfreactor-loader' ]
         }
      ]
   },
   plugins: [
      new PdfReactorServerPlugin( {
         port: 8099
      } )
   ]
};
```
